src/processing/ParameterizedOperationsUtil.java

```diff
@@ -3,195 +3,404 @@
+
+		BoardObject dummyObj = null;
+		try {
+
+			// remove previous object from maps.
+			dummyObj = ClientBoardState.maps.removeObjectFromMaps(obj.getObjectId());
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS, 
+				"colorChangeU: old object deleted"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR, 
+				"colorChangeU: object deletion failed!"
+			);
+		}
+
+		BoardObject newObj = null;
+		try {
+
+			// create a new object with same objectId, timestamp and other updated values.
+			newObj = CurveBuilder.drawCurve(newPixelSet, newBoardOp, newObjectId, newTime,
+			id, prevPixelIntensities, false);
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS, 
+				"colorChangeU: object recreated"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR, 
+				"colorChangeU: object recreation failed!"
+			);
+		}
+
+		return newObj;
+	}
+
+	/**
+	 * Function to rotate the object, store all required object's data members in local
+	 * variables, call for removing previous object from all utility maps, and finally create
+	 * new updated object.
+	 * To be called by ClientSideProcessing(via rotation function) and UndoRedo sections.
+	 */
+	public static BoardObject rotationUtil(BoardObject obj, UserId id, Angle angleOfRotation) {
+
+		// storing initial list of pixels (to be used in creation of new BoardObject)
+		ArrayList<Pixel> prevPixelIntensities = new ArrayList<Pixel>();
+		prevPixelIntensities = obj.getPixels();
+
+		Position centre = null;
+		try {
+
+			// finding centre of rotation (i.e centre Position of previous pixels)
+			centre = new Position(findCentre(prevPixelIntensities));
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS, 
+				"rotationU: object centre found"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR, 
+				"rotationU: centre couldn't be found!"
+			);
+		}
+
+		double[][] rotMatrix = new double[2][2];
+		try {
+
+			// Calculating the required rotation matrix
+			rotMatrix = computeRotationMatrix(angleOfRotation);
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS, 
+				"rotationU: rotation matrix created"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR, 
+				"rotationU: rotation matrix couldn't be formed!"
+			);
+		}
+
+		// generating new list of object's pixels
+		ArrayList<Pixel> newPixelSet = new ArrayList<Pixel>();
+		newPixelSet = obj.getPixels();
+		for (int i = 0; i < newPixelSet.size(); i++) {
+
+			// shifting origin
+			Position posWithShiftedOrigin = new Position(newPixelSet.get(i).position.r - centre.r,
+			newPixelSet.get(i).position.c - centre.c);
+
+			// applying rotation matrix followed by Narrowing Type Casting
+			Position rotatedPos = new Position((int)(rotMatrix[0][0] * posWithShiftedOrigin.r +
```

Comment:
Can you indent it in the one argument per line style? It will look much cleaner.

Reply:
done